src/BinSkim.Rules/DwarfRules/BA3003.EnableStackProtector.cs

```diff
@@ -64,70 +71,123 @@
                 {
-                    if (ElfUtility.GetDwarfCommandLineType(info.CommandLine) != DwarfCommandLineType.Gcc)
-                    {
-                        continue;
-                    }
+                    continue;
+                }
 
-                    bool failed = false;
-                    if ((!info.CommandLine.Contains("-fstack-protector-all", StringComparison.OrdinalIgnoreCase)
-                        && !info.CommandLine.Contains("-fstack-protector-strong", StringComparison.OrdinalIgnoreCase))
-                        || info.CommandLine.Contains("-fno-stack-protector", StringComparison.OrdinalIgnoreCase))
-                    {
-                        failed = true;
-                    }
-                    else
-                    {
-                        string[] paramToCheck = { "--param=ssp-buffer-size=" };
-                        string paramValue = string.Empty;
-                        bool found = GetOptionValue(info.CommandLine, paramToCheck, OrderOfPrecedence.FirstWins, ref paramValue);
+                bool failed = false;
+                if ((!info.CommandLine.Contains("-fstack-protector-all", StringComparison.OrdinalIgnoreCase)
+                    && !info.CommandLine.Contains("-fstack-protector-strong", StringComparison.OrdinalIgnoreCase))
+                    || info.CommandLine.Contains("-fno-stack-protector", StringComparison.OrdinalIgnoreCase))
+                {
+                    failed = true;
+                }
+                else
+                {
+                    string[] paramToCheck = { "--param=ssp-buffer-size=" };
+                    string paramValue = string.Empty;
+                    bool found = GetOptionValue(info.CommandLine, paramToCheck, OrderOfPrecedence.FirstWins, ref paramValue);
 
-                        if (found && !string.IsNullOrWhiteSpace(paramValue))
+                    if (found && !string.IsNullOrWhiteSpace(paramValue))
+                    {
+                        if (int.TryParse(paramValue, out int bufferSize))
                         {
-                            if (int.TryParse(paramValue, out int bufferSize))
+                            if (bufferSize > 4)
                             {
-                                if (bufferSize > 4)
-                                {
-                                    failed = true;
-                                }
+                                failed = true;
                             }
                         }
                     }
+                }
 
-                    if (failed)
+                if (failed)
+                {
+                    failedList.Add(info);
+                }
+            }
+
+            return !failedList.Any();
+        }
+
+        private static bool AnalyzeSymbols(ElfBinary binary)
+        {
+            foreach (ISection section in binary.ELF.Sections)
+            {
+                if (section.Type == SectionType.DynamicSymbolTable)
```

Comment:
>var symbols = section as SymbolTable<ulong>; [](http://example.com/codeflow?start=20&length=44)

LGTM alret 2:
Variable `symbols` may be null, when used below. Some why of check and deal with null is needed.

Reply:
Fixed